core/src/main/scala/kanaloa/reactive/dispatcher/Dispatcher.scala

```diff
@@ -55,12 +55,15 @@ object Dispatcher {
   case class Settings(
     workTimeout:     FiniteDuration                 = 1.minute,
     workRetry:       Int                            = 0,
+    updateInterval:  FiniteDuration                 = 1.second,
     dispatchHistory: DispatchHistorySettings,
     workerPool:      ProcessingWorkerPoolSettings,
     circuitBreaker:  Option[CircuitBreakerSettings],
     backPressure:    Option[BackPressureSettings],
     autoScaling:     Option[AutoScalingSettings]
-  )
+  ) {
+    val metricsCollectorSettings = MetricsCollector.MetricsCollectorSettings(updateInterval)
```

Comment:
`MetricsCollector` should be an internal concept and the settings should be derived from other public settings. The `sampleRate` should be synced with `updateInterval` and the `thresholdOfIdleWorkers` should be set by the dispatcher based on their mechanism (pulling vs pushing) . The only thing that allows some wiggling is that "minSampleDurationRatio", but I feel that should be changed by user either.